tools/confix/cmd/mutate.go

```diff
@@ -56,6 +62,14 @@ func SetCommand() *cobra.Command {
 							return errors.New("failed to set value")
 						}
 
+						if key[0] == "home-dir" {
```

Comment:
I understand the reason for this, but this feels a bit hacky.
Given that setting this value needs to do more than simply editing a file, could we add a comment above that? Or possibly actual not support it when using set.
I am actually more for doing a mea cupla and saying the set command should not support changing the home directory. Otherwise, this will cause unexpected if let's say, for whatever reason, an app adds an `home-dir` key to their app.toml and uses confix.